### DISCUSSION TITLE
How to support publish for mirror site?

### DISCUSSION THREAD

**Discussion by ihciah** [rust-lang/crates.io#4631] [General]
We run a mirror site(rsproxy.cn) for users inside China(here network status is not well for some cdn). Recently I want to support users publishing crates with the proxy.

However, I encountered 2 problems:
1. Cargo shows warning:
```
warning: using `registry.token` config value with source replacement is deprecated
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/xxx>.
Use the --token command-line flag to remove this warning.
```

But crates will be published successfully.

2. If the crate has dependency(for example parking_lot), it will fail with:
```
error: failed to publish to registry at https://rsproxy.cn

Caused by:
  the remote server responded with an error: Dependency `parking_lot` is hosted on another registry. Cross-registry dependencies are not permitted on crates.io.
```

So I want to ask if there are some plans to support publishing via mirror?

Thanks!

**Comment by Turbo87** [comment]
> So I want to ask if there are some plans to support publishing via mirror?

AFAIK there are currently no such plans
